File: shopapp-backend/shopapp/src/main/java/com/project/shopapp/model/Order.java
package com.project.shopapp.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Date;

@Data
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "fullname", length = 100)
    private String fullName;

    @Column(name = "email", length = 100)
    private String email;

    @Column(name = "phone_number", length = 200, nullable = false)
    private String phoneNumber;

    @Column(name = "address", length = 200, nullable = false)
    private String address;

    @Column(name = "note", length = 100)
    private String note;

    @Column(name = "order_date")
    private LocalDateTime orderDate;

    private String status;

    private Float totalMoney;

    @Column(name = "shipping_method", length = 100)
    private String shippingMethod;

    @Column(name = "shipping_address", length = 100)
    private String shippingAddress;

    @Column(name = "shipping_date")
    private Date shippingDate;

    @Column(name = "tracking_number", length = 100)
    private String trackingNumber;

    @Column(name = "payment_method", length = 100)
    private String paymentMethod;

    private Boolean active; //Thực hiện với admin

    @ManyToOne
    @JoinColumn(name = "user_id ")
    private User user;
}

File: angular-frontend/src/app/models/order.ts
import { OrderDetail } from './order.detail';

export interface Order {
  id: number;
  fullname: string;
  email: string;
  phone_number: string;
  address: string;
  note: string;
  order_date: Date;
  status: string;
  total_money: number;
  shipping_method: string;
  shipping_address: string;
  shipping_date: Date | null; // Có thể null nếu chưa giao hàng
  tracking_number: string | null; // Có thể null nếu chưa có mã vận chuyển
  payment_method: string;
  active: boolean; // Chỉ dùng với admin
  orderDetails: OrderDetail[];
}
